e2e-tests/test_client.py

```diff
@@ -246,4 +247,62 @@ def test_list_all_ns(obj_name):
 
     finally:
         client.delete(Namespace, name=ns1)
-        client.delete(Namespace, name=ns2)
\ No newline at end of file
+        client.delete(Namespace, name=ns2)
+
+
+@pytest.mark.parametrize("resource", [Node])
+def test_wait_global(resource):
+    client = Client()
+
+    for obj in client.list(resource):
+        client.wait(resource, obj.metadata.name, ["Ready"])
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("resource", [Node])
+async def test_wait_global_async(resource):
+    client = AsyncClient()
+
+    async for obj in client.list(resource):
+        await client.wait(resource, obj.metadata.name, ["Ready"])
+
+    await client.close()
+
+
+@pytest.mark.parametrize(
+    "resource,for_condition", [(Pod, "Ready"), (Deployment, "Available")]
+)
+def test_wait_namespaced(resource, for_condition):
```

Comment:
This test seems to fail randomly (see CI) or take a lot of time (condition never met?). I suggest to wait for an object under control of the test. For example we can create a pod (like in other tests) and wait for it to be ready using client.wait().

Reply:
@gtsystem Good call, updated. That also caught a bug where `status.conditions` can be empty if you create an object then wait for it right away.